core/src/main/scala/com/dwolla/security/crypto/Armor.scala

```diff
@@ -0,0 +1,33 @@
+package com.dwolla.security.crypto
+
+import cats.effect.*
+import cats.syntax.all.*
+import fs2.*
+import fs2.io.{readOutputStream, writeOutputStream}
+import org.bouncycastle.bcpg.*
+
+import java.io.*
+import scala.annotation.nowarn
+
+trait Armor[F[_]] {
+  def armor(chunkSize: ChunkSize): Pipe[F, Byte, Byte]
+
+  final def armor: Pipe[F, Byte, Byte] = armor(defaultChunkSize)
```

Comment:
I've not seen this way of writing a trait before, is the idea here to declare a function signature and then declare the default?

I guess this is how `armor` works but `armor()` wouldn't, because there isn't a default argument for the first function.

Reply:
Yep, basically! Default arguments are bad for binary compatibility, because it's hard to change the functions and make sure the compiler generates all the same synthetic methods. Writing out the different permutations of methods gives you the control needed to ensure the method names are stable across versions.

We could write `final def armor(): …` but I liked it better stylistically to omit the parens. 🤷‍♂️